tasks/main.yml

```diff
@@ -18,13 +36,124 @@
     - rsyslog
     - files
 
-- name: "Set forwarding to logstash"
+- name: "Set forwarding to logstash (without TLS)"
   lineinfile: >
     dest=/etc/rsyslog.conf
     regexp="^\*\.\* @{{ hostvars[logstash_forwarder]['private_ip']|default(hostvars[logstash_forwarder]['ansible_ssh_host']) }}:{{ logstash_syslog_port|default('514') }}"
     line="*.* @{{ hostvars[logstash_forwarder]['private_ip']|default(hostvars[logstash_forwarder]['ansible_ssh_host']) }}:{{ logstash_syslog_port|default('514') }}"
     state=present
-  when: rsyslog_conf.stat.exists|bool and logstash_forwarder is defined
+  when: use_rsyslog_tls|bool == false and rsyslog_conf.stat.exists|bool and logstash_forwarder is defined
+  notify:
+    - Restart rsyslog
+  tags:
+    - rsyslog
+    - files
+
+- name: "Set forwarding (with TLS). Download papertrail-bundle.pem"
+  get_url: >
+    url='https://papertrailapp.com/tools/papertrail-bundle.pem'
+    dest="/etc/pki/tls/certs/papertrail-bundle.pem"
+    mode=0444
+  when: use_rsyslog_tls|bool == true and rsyslog_conf.stat.exists|bool and logstash_forwarder is defined
+  notify:
+    - Restart rsyslog
+  tags:
+    - rsyslog
+    - files
+
+- name: 'Set forwarding (with TLS). Get md5sum of papertrail-bundle.pem (0 of 4)"'
+  shell: md5sum /etc/pki/tls/certs/papertrail-bundle.pem | cut -f 1 -d " "
+  register: papertrailmd5
+  when: use_rsyslog_tls|bool == true
+  tags:
+    - rsyslog
+    - files
+
+- debug: var=papertrailmd5
+  tags:
+    - rsyslog
+    - files
+
+- name: 'Set forwarding (with TLS). Verify checksum of /etc/pki/tls/certs/papertrail-bundle.pem (with TLS)'
+  fail:
+    msg: "Checksum of /etc/pki/tls/certs/papertrail-bundle.pem does not match ba3b40a34ec33ac0869fa5b17a0c80fc"
+  when: use_rsyslog_tls|bool == true and papertrailmd5.stdout != 'ba3b40a34ec33ac0869fa5b17a0c80fc'
+  tags:
+    - rsyslog
+    - files
+
+- name: "Set forwarding (with TLS) in /etc/rsyslog.conf. 1"
```

Comment:
Please use a template and put all the following lines together in the same file.